pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -268,6 +268,13 @@ type mountedPod struct {
 	// mapper used to block volumes support
 	blockVolumeMapper volume.BlockVolumeMapper
```

Comment:
yeah, I would prefer that - but there is a minor catch. For attachable volume types, I think volume entry is removed from `podVolumes` after volume is unmounted. We need a place where volume spec can still be found for device to be unmounted.

Reply:
Each `attachedVolume` has a list of `mountedPods` and each mounted pod should have a spec right?

Reply:
Sorry I used wrong field name. I meant that mountedPod entry is removed from `attachedVolume.mountedPods` entry after unmounting volume - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/volumemanager/cache/actual_state_of_world.go#L326

But we still have to unmount the device and that call will not have access to volume spec anymore.

Reply:
I am not sure if that is true. #58177 only touches `VolumeUnmount` functions - it does not yet make any modifications to `UnmountDevice` functions. 

And if you see first line of `UnmountDevice` function - https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_executor.go#L782 , it relies on `volume.Spec` present in `attachedVolume` struct. So if we remove the spec from `attachedVolume` and also `mountedPod` entry is removed because of previously successful `UnmountVolume` then how is this function going to get `volume.Spec` ?

Reply:
@gnufied yes, I missed that part. Based on this and also attach/detach issue, I think it would be better to file a follow up issue and let this merge first. 
@saad-ali are you ok with it?